### PR TITLE
Replace direct OpenAPI extension reference to add `x-ms-azure-resource` with internal decorator

### PR DESCRIPTION
**Pull request by allenjzhang**
Removed direct reference to OpenAPI extension `x-ms-azure-resource` in ARM library and replaced with `@Azure.ResourceManager.Private.azureResourceBase` decorator. It is only used internally on base resource types. `autorest` emitter has been updated to check the decorator and still emit `x-ms-azure-resource` extension in swagger.

### CODE REVIEW

File: .chronus/changes/azhang_RemoveAzureResource-2024-6-3-12-45-15.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+changeKind: internal
```
Comment (timotheeguerin) [inline .chronus/changes/azhang_RemoveAzureResource-2024-6-3-12-45-15.md:2]: doesn't feel like an internal change

File: packages/typespec-azure-resource-manager/src/private.decorators.ts [hunk 2]
```diff
@@ -350,3 +350,12 @@ function hasProperty(program: Program, model: Model): boolean {
   if (model.baseModel) return hasProperty(program, model.baseModel);
   return false;
 }
+
+export function $azureResourceBase(context: DecoratorContext, resourceType: Model) {
+  context.program.stateMap(ArmStateKeys.azureResourceBase).set(resourceType, true);
+}
+
+export function isAzureResourceBase(program: Program, resourceType: Model): boolean {
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/src/private.decorators.ts:358]: isn't it `isAzureResource` more accuratE?>

Comment (chrisradek) [reply packages/typespec-azure-resource-manager/src/private.decorators.ts:358]: +1 - not sure what the distinction would be with `base`

File: .chronus/changes/azhang_RemoveAzureResource-2024-6-3-12-45-15.md (hunk 1 above)
Comment (allenjzhang) [reply .chronus/changes/azhang_RemoveAzureResource-2024-6-3-12-45-15.md:2]: As we expect all ARM spec writers to extend from `Tracked/Proxy` or extreme cases `Resource`, spec writer should not use this nor the need to add `@extension("x-ms-azure-resource", true)` themselves. Hence I placed it in the `Private` namesapce. With the pending linting rule on `no reference Private namespace`, it is a pure internal matter, from outside (spec write et al) looking in, No?

Comment (timotheeguerin) [reply .chronus/changes/azhang_RemoveAzureResource-2024-6-3-12-45-15.md:2]: i mean this expose a new api, if that change is internal this means it shouildn't technically rquire a new release which isn't accurate here

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**